pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4734,6 +4735,27 @@ TRACE_ENT_FUNC(do_splice, DIRTY_PIPE_SPLICE);
 SEC("kretprobe/do_splice")
 int BPF_KPROBE(trace_ret_do_splice)
 {
+    if (!event_chosen(DIRTY_PIPE_SPLICE))
+            return 0;
+
+    // The Dirty Pipe vulnerability exist in the kernel since version 5.8, so there is not use to do logic if version
+    // is too old. In non-CORE, it will even mean using defines which are not available in the kernel headers, which
+    // will cause bugs.
+    #if !defined(CORE) && (LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0))
+    return 0;
+    #else
+    #ifdef CORE
+    // Check if field of struct exist to determine kernel version - some fields
+    // change between versions.
+    // In version 5.8 of the kernel, the field "high_zoneidx" changed its name to "highest_zoneidx".
+    // This means that the existence of the field "high_zoneidx" can indicate that the kernel version is lower
+    // than v5.8
+    struct alloc_context *check_508;
+    if (bpf_core_field_exists(check_508->high_zoneidx)) {
```

Comment:
@yanivagman I agree with you this is "hacky". If you think we can wait until the "per event" support I'm okay. Please also, take a look at recent merges using the same technique (this goes together with that older discussion os us about not being able to branch condition kconfig relocations, thus the use of this "technique"). Maybe we should revisit those as well.